ui/app/components/secret-engine/configuration-details.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { allEngines } from 'vault/helpers/mountable-secret-engines';
+
+import type Store from '@ember-data/store';
+import type SecretEngineModel from 'vault/models/secret-engine';
+import type AdapterError from '@ember-data/adapter';
+import type Model from '@ember-data/model';
+
+/**
+ * @module ConfigurationDetails
+ * `ConfigurationDetails` is used by configurable secret engines (AWS, SSH) to show either an API error, configuration details, or a prompt to configure the engine. Which of these is shown is determined by the engine type and whether the user has configured the engine yet.
+ *
+ * @example
+ * ```js
+ * <SecretEngine::ConfigurationDetails @model={{this.model}} />
+ * ```
+ *
+ * @param {object} model - The secret-engine model to be configured.
+ */
+
+interface Args {
+  model: SecretEngineModel;
+}
+
+interface ConfigError {
+  httpStatus: number | null;
+  message: string | null;
+  errors: object | null;
+}
+
+export default class ConfigurationDetails extends Component<Args> {
+  @service declare readonly store: Store;
+  @tracked configError: ConfigError | null = null;
+  @tracked configModel: Model | null = null;
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+    const { model } = this.args;
+    // Should not be able to get here without a model, but in case an upstream change allows it, catching the failure.
+    if (!model) {
+      this.configError = {
```

Comment:
I would think that if we do not expect this component to handle the error screen (it's handled upstream) then no model would default to the empty state. In that case, we could just return here without setting a tracked configError value